packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -423,25 +422,31 @@ export class AngularCompilerPlugin {
   }
 
   private _listLazyRoutesFromProgram(): LazyRouteMap {
-    let lazyRoutes: LazyRoute[];
+    let entryRoute: string | undefined;
+    let ngProgram: Program;
+
     if (this._JitMode) {
       if (!this.entryModule) {
         return {};
       }
 
-      const ngProgram = createProgram({
+      time('AngularCompilerPlugin._listLazyRoutesFromProgram.createProgram');
+      ngProgram = createProgram({
         rootNames: this._rootNames,
         options: { ...this._compilerOptions, genDir: '', collectAllErrors: true },
         host: this._compilerHost,
       });
+      timeEnd('AngularCompilerPlugin._listLazyRoutesFromProgram.createProgram');
 
-      lazyRoutes = ngProgram.listLazyRoutes(
-        this.entryModule.path + '#' + this.entryModule.className,
-      );
+      entryRoute = this.entryModule.path + '#' + this.entryModule.className;
     } else {
-      lazyRoutes = (this._program as Program).listLazyRoutes();
+      ngProgram = this._program as Program;
     }
 
+    time('AngularCompilerPlugin._listLazyRoutesFromProgram.listLazyRoutes');
+    const lazyRoutes = ngProgram.listLazyRoutes(entryRoute);
```

Comment:
Can you add a comment here that in AOT,  the entry point isn’t used and all routes within the program are returned regardless of entry point?

Reply:
done.